cst438_gradebook/src/test/java/com/cst438/AssignmentControllerTests.java

```diff
@@ -0,0 +1,184 @@
+
+import java.sql.Date;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class AssignmentControllerTests {
+    @Autowired
+    private MockMvc mvc;
+
+    @Mock
+    private AssignmentRepository assignmentRepository;
+    @Mock
+    private CourseRepository courseRepository;
+
+    @Test
+    public void DeleteAssignmentTest() throws Exception {
+        int assignmentId = 1;
+        Assignment currentAssign = new Assignment();
+        currentAssign.setId(assignmentId);
+
+        when(assignmentRepository.findById(assignmentId)).thenReturn(Optional.of(currentAssign));
+
+        mvc.perform(delete("/assignment/{assignment_id}", assignmentId)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isBadRequest());
+    }
+
+
+    @Test
+    public void GetAssignmentByIdTest() throws Exception {
+        int assignmentId = 1;
+        Assignment assignment = new Assignment();
+        assignment.setId(assignmentId);
+        assignment.setName("software");
+        when(assignmentRepository.findById(assignmentId)).thenReturn(Optional.of(assignment));
+
+        MockHttpServletResponse response;
+        response = mvc.perform(MockMvcRequestBuilders
+                        .get("/assignment/{assignment_id}", assignmentId)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andReturn().getResponse();
+
+        assertEquals(200, response.getStatus());
+    }
+    @Test
+    public void AddAssignmentTest() throws Exception {
+        AssignmentDTO assignmentDTO = new AssignmentDTO(1, "logic", "2023-09-28", "BUS 203 - Financial Accounting", 30157);
+
+        Assignment new_assignment = new Assignment();
+        new_assignment.setName(assignmentDTO.assignmentName());
+        new_assignment.setDueDate(Date.valueOf(assignmentDTO.dueDate()));
+        when(assignmentRepository.save(any(Assignment.class))).thenReturn(new_assignment);
+
+        MockHttpServletResponse response;
+        response = mvc.perform(MockMvcRequestBuilders.post("/assignment")
+                        .content(asJsonString(assignmentDTO))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andReturn().getResponse();
+
+        assertEquals(200, response.getStatus());
```

Comment:
It might be best practice to return a DTO rather then void, in order to use the conversion json functions to explicitly assert that the added assignments are present and match what was created, as exampled in ScheduleController